### ISSUE TITLE
Geth works incorrectly with large values of networkId

### ISSUE THREAD

**Issue by eugene-babichenko** [ethereum/go-ethereum#14359]
#### System information

Geth version: `1.6.0-stable-facc47cb`
OS & Version: Linux Mint 18.1 (4.4.0-53-generic)

#### Expected behaviour
Network ID is said to be a 64 bit unsigned integer, so if the network ID fits this requirement, everything should be working correctly.

#### Actual behaviour
Adding peers (through --bootnodes key or through admin.addPeer in the console) becomes impossible when networkId doesn't fit signed int. I reviewed the actual master branch and it seems like there is an inconsistency in the type of networkId in different places (I've seen uint64, uint32 and int as well).

#### Steps to reproduce the behaviour
1. Launch geth with a large networkId value (I used `geth --port 30303 --datadir data --networkid 48314142015 --ipcdisable --nodiscover --verbosity 6 console`).
1. Start a new node on an another port: `geth --port 30304 --datadir data_second --networkid 48314142015 --ipcdisable --nodiscover --verbosity 6 console`
1. From the new node connect to the first node: `admin.addPeer("enode://6e5d0ca547f9cf48185da4c7821222c5a0dd1715f9123634f28ab01abf96225a3813e5990bdd227cb6b3a933440091859efb39fe0ba0610447bb9f011723750d@[::]:30303?discport=0")`

#### Backtrace

````
TRACE[04-20|14:24:10] New dial task                            task="staticdial 6e5d0ca547f9cf48 127.0.0.1:30303"
DEBUG[04-20|14:24:10] Adding p2p peer                          id=6e5d0ca547f9cf48 name=Geth/v1.6.0-stable-f... addr=127.0.0.1:30303 peers=1
TRACE[04-20|14:24:10] Dial task done                           task="staticdial 6e5d0ca547f9cf48 127.0.0.1:30303"
TRACE[04-20|14:24:10] New dial task                            task="wait for dial hist expire (29.999996337s)"
TRACE[04-20|14:24:10] Starting protocol eth/63                 id=6e5d0ca547f9cf48 conn=staticdial
DEBUG[04-20|14:24:10] Ethereum peer connected                  id=6e5d0ca547f9cf48 conn=staticdial name=Geth/v1.6.0-stable-facc47cb/linux-amd64/go1.8.1
DEBUG[04-20|14:24:10] Ethereum handshake failed                id=6e5d0ca547f9cf48 conn=staticdial err="NetworkId mismatch - 1069501759 (!= 48314142015)"
TRACE[04-20|14:24:10] Protocol eth/63 failed                   id=6e5d0ca547f9cf48 conn=staticdial err="NetworkId mismatch - 1069501759 (!= 48314142015)"
DEBUG[04-20|14:24:10] Removing p2p peer                        id=6e5d0ca547f9cf48 conn=staticdial duration=658.15µs peers=0 req=false err="NetworkId mismatch - 1069501759 (!= 48314142015)"
````

**Comment by karalabe** [comment]
Uff, thanks for reporting this. I'll fix it for Geth 1.6.1.

**Comment by karalabe** [comment]
Could you check if https://github.com/ethereum/go-ethereum/pull/14377 solves the issue?
